acvm/src/lib.rs

```diff
@@ -74,7 +79,11 @@ pub trait PartialWitnessGenerator {
                         Self::solve_black_box_function_call(initial_witness, bb_func)
                     }
                     Opcode::Directive(directive) => {
-                        Self::solve_directives(initial_witness, directive)
+                        Self::solve_directives(initial_witness, directive).map(|possible_log| {
```

Comment:
I don't think the solve should take care of the logs, this should be handled internally by the log directive.

Reply:
I can move this logic to be internal to `solve_directives`